MainWindow.cpp

```diff
@@ -45,17 +52,32 @@ MainWindow::MainWindow(QWidget *parent) : QMainWindow(parent), ui(new Ui::MainWi
   ui->mdiArea->setBackground(QImage(":/banner.png"));
   connect(ui->menuWindow, &QMenu::aboutToShow, this, &MainWindow::updateWindowMenu);
 
+  auto settingsButton = static_cast<QToolButton *>(ui->mainToolBar->widgetForAction(ui->actionSettings));
+  settingsButton->setPopupMode(QToolButton::ToolButtonPopupMode::MenuButtonPopup);
+  settingsButton->setToolButtonStyle(Qt::ToolButtonStyle::ToolButtonTextBesideIcon);
+  ui->actionSettings->setMenu(ui->menuChangeGameSettings);
+
   RGMPlugin *pluginServer = new ServerPlugin(*this);
   auto outputTextBrowser = this->ui->outputTextBrowser;
-  connect(pluginServer, &RGMPlugin::OutputRead, outputTextBrowser, &QTextBrowser::append);
-  connect(pluginServer, &RGMPlugin::ErrorRead, outputTextBrowser, &QTextBrowser::append);
+  connect(pluginServer, &RGMPlugin::LogOutput, outputTextBrowser, &QTextBrowser::append);
+  connect(pluginServer, &RGMPlugin::CompileStatusChanged, [=](bool finished) {

```

Comment:
I'd move this lambda into the compile plugin territory, or something, as an actual function. That said, I'm pleased this MainWindow source is only a few hundred lines; I was afraid it would grow without bound.

Reply:
Well the idea behind having the lambda here is because if somebody makes a plugin for a different compiler, say Dejavu, then the behavior should work right out of the box as far as the editing state the IDE is in goes (e.g, the Run button being enabled or disabled). But yeah, I don't expect the main window class to get much bigger either. We've definitely done a better job of organizing here than LateralGM.

Reply:
Also, let me add that I did it this way so I didn't have to expose the run/debug/compile action/buttons to plugins.